DatadogLogs/Sources/Feature/Baggages.swift

```diff
@@ -0,0 +1,75 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2019-Present Datadog, Inc.
+ */
+
+import Foundation
+import DatadogInternal
+
+/// Error message sent from Logs on the message-bus.
+internal struct ErrorMessage: Encodable {
+    static let key = "error"
+    /// The Log error message
+    let message: String
+    /// The Log error type
+    let type: String?
+    /// The Log error stack
+    let stack: String?
+    /// The Log error stack
+    let source: String = "logger"
+    /// The Log attributes
+    let attributes: AnyEncodable
+}
+
+/// The Span context received from `DatadogCore`.
+internal struct SpanContext: Decodable {
+    static let key = "span_context"
+
+    enum CodingKeys: String, CodingKey {
+        case traceID = "dd.trace_id"
+        case spanID = "dd.span_id"
+    }
+
+    let traceID: String?
+    let spanID: String?
+
+    var internalAttributes: [String: String?] {
+        [
+            CodingKeys.traceID.rawValue: traceID,
+            CodingKeys.spanID.rawValue: spanID
+        ]
+    }
+}
+
+/// The RUM context received from `DatadogCore`.
```

Comment:
❓ 

Just wondering why do we have 3 `RUMContext` types? Can they be shared so we don't have to maintain it at 3 different places. 

Interestingly, two are same type but has `server_time_offset`.

```swift
    enum CodingKeys: String, CodingKey {
        case applicationID = "application.id"
        case sessionID = "session.id"
        case viewID = "view.id"
        case userActionID = "user_action.id"
    }
```

```swift
    enum CodingKeys: String, CodingKey {
        case applicationID = "application.id"
        case sessionID = "session.id"
        case viewID = "view.id"
        case viewServerTimeOffset = "server_time_offset"
    }
```

```swift
    enum CodingKeys: String, CodingKey {
        case applicationID = "application.id"
        case sessionID = "session.id"
        case viewID = "view.id"
        case userActionID = "user_action.id"
    }
```

One potential way could be the same way we forward types from `Internal` package to `RUM` and `Trace` packages. This is low effort and high value solution until we figure out something that is more elegant solution.

Reply:
That's what we are avoiding with `FeatureBaggage`, we are keeping the `Internal` module feature-agnostic. The `FeatureBaggage` allows to share encodable objects on the message-bus and in the core context, it's a loose contract between features. Types are defined end-to-end and objects are decoded to match the expected definition. 

We have some definitions that are part of the `Internal` module that comply with `PassthroughAnyCodable`, those types can be transmitted as is, without encoding/decoding, as they are known by multiple parties. But the `RUMContext` is not one of them. I believe this is the elegant solution you are referring to?

Reply:
IMO, a package with just APIs makes more sense but that would make our package count x2 (elegant solution)
eg, TraceAPI and TraceSDK (names are for illustrations only).

I don't see any benefit of loose coupling here. Correct me if I'm wrong here. Our packages don't work with mixed versions like `Rum version X` and `Trace version Y`. They have to be same because that's how our repo structured. With current "loose coupling" we are intentionally increasing our work of maintaining 2/3 or 4 types. With sharing types, they are still loose coupled - there is no direct dependency but provide strongly typed nature.

Reply:
Please have a look to #1112 and #1113 for more context on this solution. The benefit is to avoid feature-specific logic and type definitions in the Internal module. The goal is to have a clear separation of concerns between modules and no shared responsibilities. I don't see the increase in maintenance, especially for a type as simple as the RUM context. Moreover, we are transmitting way more complex types, e.g RUM View with Crash Reporting, where this solution with type-erasure drastically simplify data transmission.

The exception to the rule is the `NetworkInstrumentation`, it was decided to share it in `DatadogInternal` for simplicity of the migration, but we should consider extracting it to its dedicated module.

Reply:
For now I think it is fine but we should revisit this soon. We are going to have more and more such cases.

Another topic for dev-chat.